src/server/server_family.cc

```diff
@@ -1109,6 +1108,24 @@ std::optional<fb2::Future<GenericError>> ServerFamily::Load(string_view load_pat
 
   auto& pool = service_.proactor_pool();
 
+  if (!expand_result->summary_file.empty()) {
```

Comment:
what is the reason to save used memory only to summary file?
if you save it to all files headers when loading you will also get fail once you start loading each file.
Now you need this extra code to load summary file first

Reply:
i agree but I thought having a singular check is cleaner than doing the same logic on data shards. Are you saying it will also fix the replication?

Reply:
also not that you write the used_memory to snapshot in this condition
save_mode_ != SaveMode::SINGLE_SHARD
for replication you will write it to SINGLE_SHARD_WITH_SUMMARY there for replication will fail on a single shard while trying to replicate all shards. 
I think failing all shard is better.
Also we need a pytest for this

Reply:
As I understand current implementation should work for replication, as one of the shards will fail resulting all shards fail, but from my pov it is cleaner code to have this in all shards header